client/src/components/SearchInput.js

```diff
@@ -1,16 +1,95 @@
 import React from 'react'
-import { Box, Button, TextInput } from 'grommet'
+import { Box, Button, Select, TextInput, Keyboard } from 'grommet'
+import { useMaterialsSearch } from '../hooks/useMaterialsSearch'
+import { Mappings } from './resources'
+import { sortedObjectKeys } from '../helpers/sortObjectKeys'
+import { getReadable } from '../helpers/readableNames'
+
+export default function SearchInput({
+  initialString = '',
+  onChange,
+  size = 'medium'
+}) {
+  const {
+    query: { search },
+    addFacet,
+    removeFacet,
+    goToSearchResults,
+    setSearchString
+  } = useMaterialsSearch()
+  const [resourceType, setResourceType] = React.useState('ALL')
+  const [string, setString] = React.useState(search || initialString)
+  const resourceOptions = [
+    { label: 'All', value: 'ALL' },
+    ...sortedObjectKeys(Mappings).map((mapping) => {
```

Comment:
This could be pulled out into a constant so I will be easier to read that you are converting the mapping keys to objects for the select to consume.